tests/tracking/_tracking_service/test_utils.py

```diff
@@ -318,3 +320,15 @@ def test_resolve_tracking_uri_with_no_param():
         default_uri = "databricks://tracking_zlkjdas"
         get_tracking_uri_mock.return_value = default_uri
         assert _resolve_tracking_uri() == default_uri
+
+
+def test_store_object_can_be_serialized_by_pickle(tmpdir):
+    """
+    This test ensures a store object generated by `_get_store` can be serialized by pickle
+    to prevent issues such as https://github.com/mlflow/mlflow/issues/2954
+    """
+    pickle.dump(_get_store(f"file://{tmpdir.join('mlflow').strpath}"), io.BytesIO())
+    pickle.dump(_get_store("databricks"), io.BytesIO())
+    pickle.dump(_get_store("https://example.com"), io.BytesIO())
+    # pickle.dump(_get_store(f"sqlite:///{tmpdir.strpath}/mlflow.db"), io.BytesIO())
+    # This throws `AttributeError: Can't pickle local object 'create_engine.<locals>.connect'`
```

Comment:
`SqlAlchemyStore` object can't be serialized by pickle because `sqlalchemy` defines the `connect` function in a function.

Reply:
Fair enough! As long as the rest store can be serialized, I think it's okay.